drafts/Asynchronous-programming-made-resilient-with-MPFT.adoc

```diff
@@ -0,0 +1,237 @@
+
+	return CompletableFuture.completedFuture("serviceA");
+
+}
+----
+
+What does this do when we call the method?
+
+Well, usually when you call a method, you call it, and then it returns a result. However, when we use the `@Asynchronous` annotation, Fault Tolerance gets in between. When you call the method, Fault Tolerance first schedules your method to run later on a different thread, then it returns you a `CompletionStage` or `Future`. This isn't the `CompletionStage` or `Future` that's returned from your method, because your method hasn't run yet, it's an object created entirely by Fault Tolerance. 
+
+image::/img/blog/FT-basic-asynchronous-execution.png[Fault Tolerance asynchronous execution flow]
+
+Sometime later, your method runs and returns a `CompletionStage` or `Future`. Now, Fault Tolerance will go and complete the `CompletionStage` or `Future` object which was returned to the caller earlier, with the value that was returned from your method. You might be wondering *_"What if the returned result is an exception? What can I do?"_*.  Read along.
+
+== How you can use `@Asynchronous`
+So `@Asynchronous` sounds great, right? But how can you use it to make your asynchronous programs resilient? Let's go through 2 use-cases:
+
+1. Applying Fault Tolerance to an API call which is already asynchronous
+2. Using the `@Asynchronous` annotation to do two things in parallel with resilience
+
+=== 1. Applying Fault Tolerance to an asynchronous API call
+One use is to apply Fault Tolerance to an asynchronous API call which returns a `CompletionStage`. Since `@Asynchronous` makes code run asynchronously <<Asynchronous-logic, as we've discussed>>, you may be wondering how this can help us with something which is already asynchronous? Well, because without it, you wouldn't be able to apply Fault Tolerance. Let's see why.
+
+Note: The reason why Fault Tolerance can only be applied to asynchronous method calls returning a `CompletionStage` and not a `Future` is https://download.eclipse.org/microprofile/microprofile-fault-tolerance-2.0.2/microprofile-fault-tolerance-spec.html#_interactions_when_returning_a_code_future_code[exaplined here].
+
+In this example, we'll use the JAX-RS client API for calling remote REST services, introduced in JAX-RS 2.1. We can build up a request to fetch a String from a given URL with a GET request and the return type is a `CompletionStage` of String:
+
+==== JAX-RS client API `CompletionStage` example:
+
+[source,java]
+----
+private Client client = ClientBuilder.newClient();
+
+public CompletionStage<String> clientDemo() {
+
+	CompletionStage<String> response = client.target("http://example.com/resource")
+					.request(MediaType.TEXT_PLAIN)
+					.rx()
+					.get(String.class);
+
+	return response;
+
+}
+----
+==== Fetching the following resource:
+[source,java]
+----
+@GET
+@Path("/resource")
+public CompletionStage<String> fetchExample() { 
+
+	return CompletableFuture.completedFuture("responseText");
+
+}
+----
```

Comment:
I don't think this is needed, the server-side code isn't relevant to the article.

If we do use it then
* "Fetching the following resource" should not be a header
* We should not have the method use the JAX-RS 2.1 asynchronous server-side API - I think that just adds confusion because it's asynchronous but not anything to do with fault tolerance.
* We should list it carefully as an _example_ of something which our client-side code _could_ be calling.

Reply:
Decided to remove the example